pkg/deploy/api/validation/validation.go

```diff
@@ -32,6 +32,9 @@ func ValidateDeploymentConfigUpdate(newConfig *deployapi.DeploymentConfig, oldCo
 	allErrs := fielderrors.ValidationErrorList{}
 	allErrs = append(allErrs, validation.ValidateObjectMetaUpdate(&newConfig.ObjectMeta, &oldConfig.ObjectMeta).Prefix("metadata")...)
 	allErrs = append(allErrs, ValidateDeploymentConfig(newConfig)...)
+	if newConfig.LatestVersion < oldConfig.LatestVersion {
```

Comment:
Upstream is adding sequence number which will eventually supersede this.  But I don't know of any reasonable case where we should allow anything other than a monotonic increment +1.  You can potentially ignore greater increments during update.

TBH I don't think this is under the user's control.  Every update of the spec should be +1.  Also, the initial creation of the deployment config should probably be 1, not 0.  But there is a lot of code here to change.  Original design is that increments of latest version are the things that result in deployments.  Changes that shouldn't result in a deployment don't increment version.  Decision about whether to increment version is not under client control.